Importing `gtc:numpy` stencils import issues on Piz Daint

Sometimes, when running `gtc:numpy`stencils on Piz Daint (SCRATCH), the follwoing type of error appears:

```
self = <gt4py.backend.gtc_backend.gtcnumpy.backend.GTCNumpyBackend object at 0x2aab22d95ac0>

    def _load(self) -> Type["StencilObject"]:
        stencil_class_name = self.builder.class_name
        file_name = str(self.builder.module_path)
        stencil_module = gt_utils.make_module_from_file(stencil_class_name, file_name)
>       stencil_class = getattr(stencil_module, stencil_class_name)
E       AttributeError: 'NoneType' object has no attribute 'two_optional_fields____gtcnumpy_412215ca76'

../src/gt4py/backend/base.py:268: AttributeError

```
however the member `two_optional_fields____gtcnumpy_412215ca76` is there. It seems to work after deleting the corresponding file from the cache and re-generating.

--- a/src/gt4py/backend/module_generator.py
+++ b/src/gt4py/backend/module_generator.py
@@ -437,13 +437,13 @@ class PyExtModuleGenerator(BaseModuleGenerator):
     and ``_has_effect()`` methods.
     """
 
-    pyext_module_name: str
-    pyext_file_path: str
+    pyext_module_name: Optional[str]
+    pyext_file_path: Optional[str]
 
     def __init__(self):
         super().__init__()
-        self.pyext_module_name = ""
-        self.pyext_file_path = ""
+        self.pyext_module_name = None
+        self.pyext_file_path = None
 
     def __call__(
         self,
@@ -456,13 +456,15 @@ class PyExtModuleGenerator(BaseModuleGenerator):
         return super().__call__(args_data, builder, **kwargs)
 
     def _is_not_empty(self) -> bool:
+        if self.pyext_module_name is None:
+            return False
         if self.builder.backend.USE_LEGACY_TOOLCHAIN:
             return iir_is_not_emtpy(self.builder.implementation_ir)
         return gtir_is_not_emtpy(self.builder.gtir_pipeline)
 
     def generate_imports(self) -> str:
         source = ["from gt4py import utils as gt_utils"]
-        if self._is_not_empty:
+        if self._is_not_empty():
             source.append(
                 textwrap.dedent(
                     f"""
@@ -475,6 +477,8 @@ class PyExtModuleGenerator(BaseModuleGenerator):
         return "\n".join(source)
 
     def _has_effect(self) -> bool:
+        if not self._is_not_empty():
+            return False
         if self.builder.backend.USE_LEGACY_TOOLCHAIN:
             return iir_has_effect(self.builder.implementation_ir)
         return gtir_has_effect(self.builder.gtir_pipeline)

--- a/src/gt4py/config.py
+++ b/src/gt4py/config.py
@@ -71,6 +71,8 @@ build_settings: Dict[str, Any] = {
 cache_settings: Dict[str, Any] = {
     "dir_name": os.environ.get("GT_CACHE_DIR_NAME", ".gt_cache"),
     "root_path": os.environ.get("GT_CACHE_ROOT", os.path.abspath(".")),
+    "load_retries": os.environ.get("GT_CACHE_LOAD_RETRIES", 3),
+    "load_retry_delay": os.environ.get("GT_CACHE_LOAD_RETRY_DELAY", 100),
 }
 
 code_settings: Dict[str, Any] = {"root_package_name": "_GT_"}

--- a/src/gt4py/config.py
+++ b/src/gt4py/config.py
@@ -72,7 +72,7 @@ cache_settings: Dict[str, Any] = {
     "dir_name": os.environ.get("GT_CACHE_DIR_NAME", ".gt_cache"),
     "root_path": os.environ.get("GT_CACHE_ROOT", os.path.abspath(".")),
     "load_retries": os.environ.get("GT_CACHE_LOAD_RETRIES", 3),
-    "load_retry_delay": os.environ.get("GT_CACHE_LOAD_RETRY_DELAY", 100), # unit miliseconds
+    "load_retry_delay": os.environ.get("GT_CACHE_LOAD_RETRY_DELAY", 100),  # unit miliseconds
 }
 
 code_settings: Dict[str, Any] = {"root_package_name": "_GT_"}

--- a/src/gt4py/config.py
+++ b/src/gt4py/config.py
@@ -72,7 +72,7 @@ cache_settings: Dict[str, Any] = {
     "dir_name": os.environ.get("GT_CACHE_DIR_NAME", ".gt_cache"),
     "root_path": os.environ.get("GT_CACHE_ROOT", os.path.abspath(".")),
     "load_retries": os.environ.get("GT_CACHE_LOAD_RETRIES", 3),
-    "load_retry_delay": os.environ.get("GT_CACHE_LOAD_RETRY_DELAY", 100),
+    "load_retry_delay": os.environ.get("GT_CACHE_LOAD_RETRY_DELAY", 100), # unit miliseconds
 }
 
 code_settings: Dict[str, Any] = {"root_package_name": "_GT_"}

--- a/src/gt4py/utils/base.py
+++ b/src/gt4py/utils/base.py
@@ -320,7 +320,7 @@ def make_module_from_file(qualified_name, file_path, *, public_import=False):
         try:
             return load()
         except ModuleNotFoundError:
-            time.sleep(max(config.cache_settings["load_retry_delay"], 0))
+            time.sleep(max(config.cache_settings["load_retry_delay"], 0) / 1000)
 
     return load()
 

--- a/src/gt4py/utils/base.py
+++ b/src/gt4py/utils/base.py
@@ -32,7 +32,7 @@ import time
 import types
 from typing import Any, Sequence, Tuple
 
-from gt4py import config
+from gt4py import config as gt_config
 
 
 NOTHING = object()
@@ -316,11 +316,11 @@ def make_module_from_file(qualified_name, file_path, *, public_import=False):
                     setattr(parent, module_name, module)
         return module
 
-    for i in range(max(config.cache_settings["load_retries"], 0)):
+    for i in range(max(gt_config.cache_settings["load_retries"], 0)):
         try:
             return load()
         except ModuleNotFoundError:
-            time.sleep(max(config.cache_settings["load_retry_delay"], 0) / 1000)
+            time.sleep(max(gt_config.cache_settings["load_retry_delay"], 0) / 1000)
 
     return load()
 

--- a/src/gt4py/utils/base.py
+++ b/src/gt4py/utils/base.py
@@ -28,8 +28,11 @@ import json
 import os
 import string
 import sys
+import time
 import types
 from typing import Any, Sequence, Tuple
+
+from gt4py import config
 
 
 NOTHING = object()
@@ -287,8 +290,11 @@ def make_module_from_file(qualified_name, file_path, *, public_import=False):
       https://stackoverflow.com/a/43602645
 
     """
-    try:
+
+    def load():
         spec = importlib.util.spec_from_file_location(qualified_name, file_path)
+        if not spec:
+            raise ModuleNotFoundError(f"No module named '{qualified_name}'")
         module = importlib.util.module_from_spec(spec)
         spec.loader.exec_module(module)
         if public_import:
@@ -308,12 +314,15 @@ def make_module_from_file(qualified_name, file_path, *, public_import=False):
                         break
                 else:
                     setattr(parent, module_name, module)
+        return module
 
-    except Exception as e:
-        print(e)
-        module = None
+    for i in range(max(config.cache_settings["load_retries"], 0)):
+        try:
+            return load()
+        except ModuleNotFoundError:
+            time.sleep(max(config.cache_settings["load_retry_delay"], 0))
 
-    return module
+    return load()
 
 
 def patch_module(module, member, new_value, *, recursive=True):

